src/edu/csus/ecs/pc2/ui/team/QuickSubmitter.java

```diff
@@ -91,6 +93,25 @@ public List<File> getAllCDPsubmissionFileNames(IInternalContest mycontest, Strin
 
         return files;
     }
+    
+    
+    public File sendSubmission(File sourceFile) throws Exception {
+        File submittedFile = null;
+
+        Language language = LanguageUtilities.guessLanguage(getContest(), sourceFile.getAbsolutePath());
+        if (language == null) {
+            String ext = LanguageUtilities.getExtension(sourceFile.getAbsolutePath());
+            throw new Exception("Cannot identify language for ext= " + ext + " = Can not send submission for file " + sourceFile.getAbsolutePath());
+        } else {
+            Problem problem = guessProblem(getContest(), sourceFile.getAbsolutePath());
```

Comment:
When this method calls `guessLanguage()` (several lines above),  it checks for a null return (which can happen if `guessLanguage()` is unable to find a corresponding language), and appropriately logs an error .
However, when `sendSubmission(File)` calls `guessProblem()` (the statement above), it does not check for a null return value. I think the code needs to log an error in that condition just like it does when `guessLanguage()` returns null.

I consider this an approval showstopper.